tests/test_neighbor_search.py

```diff
@@ -0,0 +1,142 @@
+    positions = np.random.rand(n_particle*3).reshape(3, n_particle)
+    for cur_class in neighbor_classes:
+        cur_instance = cur_class(interaction_distance=0.3, interaction_depth=0.3)
+        cur_instance.rebuild(positions)
+        instances.append(cur_instance)
+
+    for particle_idx in np.random.choice(positions.shape[1], 100, replace=False):
+        compare_results_by_idx(instances, particle_idx)
+
+
+def create_spherical_positions(n_particles, max_depth=100000):
+    yrange = 2*np.random.rand(n_particles)
+    lat = 180*(np.arccos(1-yrange)-0.5*np.pi)/np.pi
+    long = 360*np.random.rand(n_particles)
+    depth = max_depth*np.random.rand(n_particles)
+    return np.array((lat, long, depth))
+
+
+def create_flat_positions(n_particle):
+    return np.random.rand(n_particle*3).reshape(3, n_particle)
+
+
+def test_spherical_neighbors():
+    np.random.seed(9837452)
+    neighbor_classes = [
+        BruteSphericalNeighborSearch, HashSphericalNeighborSearch
+    ]
+
+    instances = []
+    positions = create_spherical_positions(10000, max_depth=100000)
+    for cur_class in neighbor_classes:
+        cur_instance = cur_class(interaction_distance=1000000, interaction_depth=100000)
+        cur_instance.rebuild(positions)
+        instances.append(cur_instance)
+
+    for particle_idx in np.random.choice(positions.shape[1], 100, replace=False):
+        compare_results_by_idx(instances, particle_idx)
+
+
+def test_flat_update():
+    np.random.seed(9182741)
+    n_particle = 1000
+    n_test_particle = 10
+    n_active_mask = 10
+    neighbor_classes = [
+        ScipyFlatNeighborSearch, BruteFlatNeighborSearch, HashFlatNeighborSearch
+    ]
+
+    instances = []
+    for cur_class in neighbor_classes:
+        cur_instance = cur_class(interaction_distance=0.3, interaction_depth=0.3)
+        instances.append(cur_instance)
+
+    for i in range(n_active_mask):
+        positions = create_flat_positions(n_particle) + 10*np.random.rand()
+        if i == 0:
+            active_mask = None
+        else:
+            active_mask = np.random.rand(n_particle) > 0.5
+        for cur_instance in instances:
+            cur_instance.update_values(positions, active_mask)
+        active_idx = np.where(active_mask)[0]
+        if len(active_idx) == 0:
+            continue
+        test_particles = np.random.choice(
+            active_idx, size=min(n_test_particle, len(active_idx)), replace=False)
+        for particle_idx in test_particles:
+            compare_results_by_idx(instances, particle_idx, active_idx=active_idx)
+
+
+def test_spherical_update():
+    np.random.seed(9182741)
+    n_particle = 1000
+    n_test_particle = 10
+    n_active_mask = 10
+    neighbor_classes = [
+        BruteSphericalNeighborSearch, HashSphericalNeighborSearch
+    ]
+
+    instances = []
+    for cur_class in neighbor_classes:
+        cur_instance = cur_class(interaction_distance=1000000, interaction_depth=100000)
+        instances.append(cur_instance)
```

Comment:
instead of putting those `neighbor_classes` inside the test function, it would be nicer to use a `pytest.parametrize(...)` setup similar to what we do for `pset_modes` and `mode` (latter-i.e.: `['scipy', 'jit']`)

Reply:
Done